src/app/pages/workspace/workspace-search-tile/workspace-search-tile.component.ts

```diff
@@ -7,9 +25,292 @@ import { Component, OnInit } from '@angular/core';
 })
 export class WorkspaceSearchTileComponent implements OnInit {
 
-  constructor() { }
+  constructor(private corpusStateService: CorpusStateService,
+    private searchService: SearchService,
+    private commonService: CommonService,
+    private loaderService: LoaderService) { }
+
+  /**initial panel size */
+  currentPanelHeight: number = 500;
+
+  /**search data parameters */
+  searchValue: string = '';
+  searchModes!: Array<SearchMode>;
+  selectedSearchMode!: SearchMode;
+  searchLabel: string = '';
+  contextLenghtDefaultValue = 5;
+  contextLength: number = this.contextLenghtDefaultValue;
+  contextMaxLenght: number = 10;
+  files!: TreeNode<CorpusElement>[];
+  selectedDocuments: TreeNode<CorpusElement>[] = [];
+
+  //**kwic table data */
+  searchResults: Array<SearchResultRow> = [];
+  searchRequest = new SearchRequest();
+  selectedSearchResults: Array<SearchResultRow> = [];
+  loading: boolean = false;
+  selectAll: boolean = false;
+  tableContainerHeight!: number;
+  tableHeaderHegith: number = 265;
+  totalRecords: number = 0;
+  visibleRows: number = 10;
+  tableCleared = false;
+  changingPage = false;
+
+  /**export button items */
+  exportItems!: MenuItem[];
+
+  @ViewChild('searchInput') searchInput: any;
+  @ViewChild('dt') searchResultsTable!: Table;
+
+  private filtersSubject: Subject<any> = new Subject();
 
   ngOnInit(): void {
+    this.corpusStateService.filesystem$.pipe(
```

Comment:
See workspace-corpus-explorer.ts ln 24
```
files$ = this.corpusStateService.filesystem$.pipe(
    switchMap(docs => of(this.mapToTreeNodes(docs))),
  );
```
Using this and an async pipe in the template, there is no need for a this.file variable (among other things, without handling the subscribe closure).

Reply:
I don't like this approach but if you like it I will change it

Reply:
I will change it